nifti2/nifti2_io_version.h

```diff
@@ -0,0 +1 @@
+#define NIFTI2_IO_VERSION "2.1.0"
```

Comment:
I definitely prefer to have separate variables, and then a macro for the string version

Additionally, a note needs to be added to all the places that need to be kept manually in sync so that future developers change all the right code.

```c
/* NOTE:  When version values change, 4 places in the code need to  be kept in sync
 CMakeLists.txt  nifti2_io_version.h nifti1_io_version.h nifticdf_version.h
*/
#define NIFTI2_VERSION_TO_STRING(x) ITK_VERSION_TO_STRING0(x)
#define NIFTI2_VERSION_TO_STRING0(x) #x
#define NIFTI2_VERSION                                                                                                    \
  NIFTI2_VERSION_TO_STRING(NIFTI2_VERSION_MAJOR)                                                                             \
  "." NIFTI2_VERSION_TO_STRING(NIFTI2_VERSION_MINOR) "." NIFTI2_VERSION_TO_STRING(NIFTI2_VERSION_PATCH)
#define NIFTI2_SOURCE_VERSION "NIFTI2 version " NIFTI2_VERSION
```